ivadomed/utils.py

```diff
@@ -236,14 +239,28 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
+            data_im_l.append(data_im_tmp)
         data_l_lst.append(data_im_l)
         del nib_im
+    data_l_lst = np.array(data_l_lst)
+    # channel went first due to the 'append' function
+    if len(data_l_lst.shape) == 4:
+        data_l_lst = np.transpose(data_l_lst, (0,2,3,4,1))
```

Comment:
i'm a bit confused here. it would be useful to add a comment on the assumed dimensions for `data_l_lst` (dim1 is X, dim2 is Y, etc.)

Reply:
Done in f4716ad , phrasing slightly changed in 51ae6d2